wlutil/build.py

```diff
@@ -73,24 +73,39 @@ def submoduleDeps(submodules):
             }
 
 
-def kmodDepsTask(cfg, name=""):
+def kmodDepsTask(cfg, taskDeps=None, name=""):
     """Check if the kernel modules in cfg are uptodate (suitable for doit's calc_dep function)"""
+
     def checkMods(cfg):
+        log = logging.getLogger()
         for driverDir in cfg['linux']['modules'].values():
-            p = run(["make", "-q", "LINUXSRC=" + str(cfg['linux']['source'])], cwd=driverDir, check=False)
+            if not driverDir.exists():
+                log.warn("WARNING: Required module " + str(driverDir) + " does not exist: Assuming the workload is not uptodate.")
+                return False
+            try:
+                p = run(["make", "-q", "LINUXSRC=" + str(cfg['linux']['source'])], cwd=driverDir, check=False)
```

Comment:
Shouldn't the subprocess already raise an error if `cwd` doesn't exist?

Reply:
Ya, it did raise an exception that wasn't being caught. I could add it to the except if I wanted, might be a bit cleaner looking. I do want to keep the two cases separate (filenotfound vs other) just for clarity to the user. I can switch it to do both in the except.

Reply:
It's fine to me as it is.  `FileNotFoundError` is a bit more general than `driverDir` not existing, for example if `make` is not found.